homeassistant/components/philips_js/light.py

```diff
@@ -0,0 +1,411 @@
+        system: dict[str, Any],
+        unique_id: str,
+    ):
+        """Initialize light."""
+        self._tv = coordinator.api
+        self._hs = None
+        self._brightness = None
+        self._system = system
+        self._unique_id = unique_id
+        self._coordinator = coordinator
+        super().__init__(coordinator)
+
+        self._update_from_coordinator()
+
+    @property
+    def name(self):
+        """Return the device name."""
+        return self._system["name"]
+
+    @property
+    def unique_id(self):
+        """Return unique identifier if known."""
+        return self._unique_id
+
+    @property
+    def supported_features(self):
+        """Return supported features on this light."""
+        if self.color_mode == COLOR_MODE_HS:
+            return SUPPORT_EFFECT | SUPPORT_COLOR | SUPPORT_BRIGHTNESS
+        return SUPPORT_EFFECT
+
+    @property
+    def should_poll(self):
+        """Device should be polled."""
+        return False
+
+    @property
+    def effect_list(self):
+        """Return the list of supported effects."""
+        effects = []
+        effects.extend(
+            _get_effect(EFFECT_AUTO, style, setting)
+            for style, data in self._tv.ambilight_styles.items()
+            if _is_valid(EFFECT_AUTO, style)
+            and _is_on(EFFECT_AUTO, style, self._tv.powerstate)
+            for setting in data.get("menuSettings", [])
+        )
+
+        effects.extend(
+            _get_effect(EFFECT_EXPERT, style, algorithm)
+            for style, data in self._tv.ambilight_styles.items()
+            if _is_valid(EFFECT_EXPERT, style)
+            and _is_on(EFFECT_EXPERT, style, self._tv.powerstate)
+            for algorithm in data.get("algorithms", [])
+        )
+
+        effects.extend(
+            _get_effect(EFFECT_MODE, style, None)
+            for style in self._tv.ambilight_modes
+            if _is_valid(EFFECT_MODE, style)
+            and _is_on(EFFECT_MODE, style, self._tv.powerstate)
+        )
+
+        return sorted(effects)
+
+    @property
+    def effect(self):
+        """Return the current effect."""
+        current = self._tv.ambilight_current_configuration
+        if current and self._tv.ambilight_mode != "manual":
+            if current["isExpert"]:
+                settings = _get_settings(current)
+                if settings:
+                    return _get_effect(
+                        EFFECT_EXPERT, current["styleName"], settings["algorithm"]
+                    )
+                return _get_effect(EFFECT_EXPERT, current["styleName"], None)
+
+            return _get_effect(
+                EFFECT_AUTO, current["styleName"], current.get("menuSetting", None)
+            )
+
+        return _get_effect(EFFECT_MODE, self._tv.ambilight_mode, None)
+
+    @property
+    def supported_color_modes(self):
+        """Return the supported color mode."""
+        return [COLOR_MODE_HS, COLOR_MODE_ONOFF]
+
```

Comment:
Replace by setting `self._supported_color_modes` in constructor. 

https://github.com/home-assistant/core/blob/dev/homeassistant/components/light/__init__.py#L603-L616